src/onefetch/info.rs

```diff
@@ -695,6 +792,61 @@ impl Info {
         };
         Some(color)
     }
+
+    fn get_formatted_subtitle_label(
+        &self,
+        label: &str,
+        color: Color,
+        colon_clr: Color,
+    ) -> ColoredString {
+        let formatted_label = format!("{}{} ", label.color(color), ":".color(colon_clr));
+        if self.config.no_bold {
+            formatted_label.normal()
+        } else {
+            formatted_label.bold()
+        }
+    }
+
+    fn get_color_set(text_colors: &[String], default_colors: &Vec<Color>) -> TextColor {
+        let mut custom_color: Vec<Color> = Vec::<Color>::new();
+        if text_colors.is_empty() {
+            custom_color.insert(0, default_colors[0]);
+            custom_color.insert(1, Color::White);
+            custom_color.insert(2, Color::White);
+            custom_color.insert(3, default_colors[0]);
+            custom_color.insert(4, default_colors[0]);
+            custom_color.insert(5, Color::White);
+        } else {
+            custom_color = text_colors
+                .iter()
+                .map(|color_num| {
+                    let custom = Info::num_to_color(color_num);
+                    match custom {
+                        Some(custom) => custom,
+                        None => Color::White,
+                    }
+                })
+                .collect();
+            if custom_color.len() < 6 {
+                for i in 0..6 {
+                    if i < custom_color.len() {
+                        continue;
+                    }
+                    custom_color.insert(i, Color::White);
+                }
+            }
+        }
+        
+        let color_set: TextColor = TextColor {
+            title: custom_color[0],
+            tilde: custom_color[1],
+            underline: custom_color[2],
+            subtitle: custom_color[3],
+            colon: custom_color[4],
+            info: custom_color[5],
+        };
```

Comment:
Looks like you're trying to enforce `custom_color` to always have 6 elements, setting every "undefined" element to `Color::White`. I think this can be simplified like so.
```diff
-title: custom_color[0],
+title: custom_color.get(0).unwrap_or(Color::White),
```
Then you don't have to insert `Color::White` into the "undefined" indices.

Reply:
Right, except that for some fields: title, subtitle, colon; we don't want Color::White but `default_colors[0]`

The implementation would look smth like:

```rust
impl TextColor {
    fn new(color: Color) -> TextColor {
        TextColor {
            title: color,
            tilde: Color::White,
            underline: Color::White,
            subtitle: color,
            colon: color,
            info: Color::White,
        }
    }
}

fn get_text_color_set(text_colors: &[String], default_colors: &Vec<Color>) -> TextColor {
    let mut text_color_set = TextColor::new(default_colors[0]);
    if !text_colors.is_empty() {
            let custom_color = text_colors
                .iter()
                .map(|color_num| {
                    let custom = Info::num_to_color(color_num);
                    match custom {
                        Some(custom) => custom,
                        None => Color::White,
                    }
                })
                .collect::<Vec<Color>>();

            text_color_set.title = custom_color.get(0).unwrap_or(default_colors[0]);
            text_color_set.tilde = custom_color.get(1).unwrap_or(Color::White);
            text_color_set.underline = custom_color.get(2).unwrap_or(Color::White);
            text_color_set.subtitle = custom_color.get(3).unwrap_or(default_colors[0]);
            text_color_set.colon = custom_color.get(4).unwrap_or(default_colors[0]);
            text_color_set.info = custom_color.get(5).unwrap_or(Color::White);
        }

    text_color_set
}
```